DC/DC support for PCA10040

carlescufi opened the pull request.

carlescufi committed:
boards: arm: nrf52_pca10040: Enable DC/DC by default

Since the PCA10040 contains the required DC/DC circuitry, enable it by
default while keeping it configurable. This reduces power consumption
during radio activity.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi committed:
boards: arm: nrf52810_pca10040: Enable DC/DC by default

Since the PCA10040 contains the required DC/DC circuitry, enable it by
default while keeping it configurable. This reduces power consumption
during radio activity.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

ioannisg reviewed (changes requested):
LGTM.

But, actually, I am wondering: why do we need to have this K-config option at all? Couldn't we just select SOC_DCDC_NRF52X in Kconfig.board,or <board>_defkconfig?
Same for nRF52840_pca10056, where the DCDC is already selected in the same way.

I would understand the K-config option, if it were used by other (nonNordic( boards. But it is not, AFAICS

cvinayak commented:
@ioannisg SOC_DCDC_NRF52X is hidden for DCDC feature's need of external components on the board, hence nRF52x SoC have them hidden only to be "select"-ed by boards (our DKs with visible BOARD_HAS_DCDC) that have the external components.

ioannisg commented:
@cvinayak ,  I don't disagree on this. Just wondering; could SOC_DCDC_NRF52X be selected directly in the boards' K-config definition?

carlescufi commented:
@ioannisg I think DC/DC is a compromise, and not everybody wants it enabled?

ioannisg commented:
OK, so all we want is to allow users to be able to un-select it, from their prj.conf, even if they are using an "official" Nordic DK, right?

carlescufi commented:
>OK, so all we want is to allow users to be able to un-select it, from their prj.conf, even if they are using an "official" Nordic DK, right?

Yes, correct

ioannisg commented:
@carlescufi understood.
Then, it seems that BOARD_HAS_DCDC is also a "user" option. I am wondering if we could write something in the comment, inside the K-config option definition or prompt (e.g. "set to -n- to remove the DCDC feature"). Otherwise, it will look that it is solely an option describing hardware capabilities (and, clearly, it is not).

nashif commented on boards/arm/nrf52810_pca10040/Kconfig:
```diff
@@ -6,4 +6,9 @@
 
 if BOARD_NRF52810_PCA10040
 
+config BOARD_HAS_DCDC
```
Kconfigs with _HAS_  should not be prompted, those should be in general hidden Kconfigs.  Also, where are those being used? You might want to change the name of the Kconfig and the prompt text .

carlescufi committed:
boards: arm: nrf52840_pca10056: Align DC/DC option

Align the name and description of the DC/DC option with the other 2
boards that use it.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

carlescufi commented:
@ioannisg @nashif fixed all comments

ioannisg commented on boards/arm/nrf52810_pca10040/Kconfig:
```diff
@@ -6,4 +6,9 @@
+        select SOC_DCDC_NRF52X
+        default y
+
+ # BOARD_NRF52810_PCA10040
```
Is there something wrong here, @carlescufi ?

ioannisg replied:
Looks in my diff that endif is missing

carlescufi replied:
Fixed, sorry.
